tests/dataframe_test.py

```diff
@@ -101,6 +101,37 @@ def test_dataframe_creation(self):
         self.assertEqual(len(df), 6)
         self.assertTrue(ref_df.equals(df.to_pandas()))
 
+    def test_client_type_creation(self):
+        f = ak.Fields(ak.arange(10), ["A", "B", "c"])
+        ip = ak.ip_address(ak.arange(10))
+        d = ak.Datetime(ak.arange(10))
+
+        df_dict = {
+            "fields": f,
+            "ip": ip,
+            "date": d
+        }
+        df = ak.DataFrame(df_dict)
+        pd_d = [pd.to_datetime(x, unit='ns') for x in d.to_list()]
+        pddf = pd.DataFrame({
+            "fields": f.to_list(),
+            "ip": ip.to_list(),
+            "date": pd_d
+        })
```

Comment:
it seems `BitVector` is missing here. I'm not sure if there's a good candidate to compare this against and it wasn't specifically requested, so I'm okay leaving it out. But I wanted to make sure it was intentionally excluded